src/render/index.js

```diff
@@ -0,0 +1,60 @@
+import echo from '../utils/console.js';
+import db, { AUCTION_STORE } from '../utils/db.js';
+import validate from '../utils/validation.js';
+import {
+	getTarget,
+	renderFrame,
+} from './utils.js';
+
+/*
+ * @function
+ * @name renderAd
+ * @description render an ad
+ * @author Newton <cnewton@magnite.com>
+ * @param {string} selector - a string reprensenting a valid selector to find an element on the page
+ * @param {string} token - a string that represents the results from an auction run via the `fledge.runAdAuction` call
+ * @throws {Error} Any parameters passed are incorrect or an incorrect type
+ * @return {Promise<null | true>}
+ *
+ * @example
+ *   renderAd('#ad-slot-1', '76941e71-2ed7-416d-9c55-36d07beff786');
+ */
+export default async function renderAd (selector, token, debug = false) {
+	debug && echo.group('Fledge: Render an Ad');
+	debug && echo.log('ad render selector:', selector);
+	debug && echo.log('ad render token:', token);
+	validate.param(selector, 'string');
+	validate.param(token, 'string');
+
+	debug && echo.info('checking that target exists on the page');
+	const target = getTarget(selector);
+	debug && echo.log('target:', target);
+	if (!target) {
+		throw new Error(`Target not found on the page! Please check that ${target} exists on the page.`);
+	}
+
+	debug && echo.info('checking that winning token exists');
+	const winner = await db.store.get(AUCTION_STORE, token);
+	debug && echo.log('winners token:', winner);
+	if (!(winner && winner.id === token)) {
+		throw new Error(`A token was not found! Token provided: ${token}`);
+	}
+
+	debug && echo.info('checking that winner to be rendered is on the same hostname as the auction');
+	if (winner?.hostname !== window.top.location.hostname) {
```

Comment:
This is a good start, I'm thinking in a latter extension we validate that the full url is the same (not just the hostname).